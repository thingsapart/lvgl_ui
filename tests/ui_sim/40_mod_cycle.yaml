# TICKS: 3
- type: data-binding
  state:
    - num_val: 1.0
    - str_val: "A"
    - bool_val: true
  actions:
    - cycle_all:
        num_val: { cycle: }
        str_val: { cycle: ["A", "B", "C"] }
        bool_val: { cycle: [true, false] }
  schedule:
    - { tick: 1, action: cycle_all }
    - { tick: 2, action: cycle_all }
    - { tick: 3, action: cycle_all }
