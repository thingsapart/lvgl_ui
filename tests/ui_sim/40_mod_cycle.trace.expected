--- UI-Sim Trace Start ---
STATE_SET: num_val = 1.000 (old: null)
STATE_SET: str_val = "A" (old: null)
STATE_SET: bool_val = true (old: null)
NOTIFY: num_val = 1.000
NOTIFY: str_val = "A"
NOTIFY: bool_val = true

--- TICK 1 ---
ACTION: cycle_all value=null
STATE_SET: num_val = 2.000 (old: 1.000)
STATE_SET: str_val = "B" (old: "A")
STATE_SET: bool_val = false (old: true)
NOTIFY: num_val = 2.000
NOTIFY: str_val = "B"
NOTIFY: bool_val = false

--- TICK 2 ---
ACTION: cycle_all value=null
STATE_SET: num_val = 3.000 (old: 2.000)
STATE_SET: str_val = "C" (old: "B")
STATE_SET: bool_val = true (old: false)
NOTIFY: num_val = 3.000
NOTIFY: str_val = "C"
NOTIFY: bool_val = true

--- TICK 3 ---
ACTION: cycle_all value=null
STATE_SET: num_val = 1.000 (old: 3.000)
STATE_SET: str_val = "A" (old: "C")
NOTIFY: num_val = 1.000
NOTIFY: str_val = "A"

--- UI-Sim Trace End ---
